File: February/CPP/Least number of unique integers/Least number of unique integers after k removals.cpp
class Solution {  
public:
  int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<int, int> countMap;
        for (int num : arr) {
            countMap[num]++;
        }

        vector<pair<int, int>> counts;
        for (auto& pair : countMap) {
            counts.push_back(pair);
        }
        sort(counts.begin(), counts.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
            return a.second < b.second;
        });

        int uniqueCount = counts.size();
        for (auto& pair : counts) {
            if (k >= pair.second) {
                k -= pair.second;
                uniqueCount--;
            } else {
                break;
            }
        }
        
        return uniqueCount;
    }
};

File: February/Python/Least Number of Unique Integers after K Removals/Least Number of Unique Integers after K Removals.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        distinct_elements = len(set(arr))

        frequencies = {}

        for num in arr:
            frequencies[num] = frequencies.get(num, 0) + 1

        for value in sorted(frequencies.values()):
            if value <= k:
                distinct_elements -= 1
                k -= value
            else:
                break

        return distinct_elements
